azext_iot/iothub/providers/state.py

```diff
@@ -213,6 +266,120 @@ def save_state(self, filename: str, overwrite_file=False):
         except FileNotFoundError:
             raise FileOperationError(f'File {filename} does not exist.')
 
+    @contextmanager
+    def capture_stderr(self):
+        temp_file = "stderr_file" + str(random.randrange(100000000))
```

Comment:
Nice idea, but what about refactoring to use an in-memory text buffer so we don't need to deal with writing/deleting files to disk https://docs.python.org/3/library/io.html#io.StringIO